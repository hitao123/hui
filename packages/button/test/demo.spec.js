import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Button from '../index.vue';
import ButtonDemo from '../demo/index.vue';

describe('Button component', () => {
  it('should show Button', () => {
    const warpper = shallowMount(Button, {
      propsData: {
        text: 'Button',
        block: false,
        plain: true,
        loading: false
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Button Demo', () => {
    expect(renderToString(ButtonDemo)).toMatchSnapshot();
  });
});
