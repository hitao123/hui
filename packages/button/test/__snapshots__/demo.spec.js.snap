// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should show Button 1`] = `
<button class="h-button h-button--default h-button--normal h-button--plain">
  <!----> <span class="h-button__text">Button</span></button>
`;

exports[`Button component should show Button Demo 1`] = `
<demo-section>
  <demo-block title="按钮类型">
    <div class="demo-button-row">
      <h-button type="default">默认按钮</h-button>
      <h-button type="primary">主要按钮</h-button>
    </div>
    <h-button type="warning">警告按钮</h-button>
    <h-button type="danger">危险按钮</h-button>
  </demo-block>
  <demo-block title="简单按钮">
    <div class="demo-button-row">
      <h-button type="default" plain="">默认按钮</h-button>
      <h-button type="primary" plain="">主要按钮</h-button>
    </div>
    <h-button type="warning" plain="">警告按钮</h-button>
    <h-button type="danger" plain="">危险按钮</h-button>
  </demo-block>
  <demo-block title="按钮大小">
    <div class="demo-button-row">
      <h-button size="large">大号按钮</h-button>
    </div>
    <h-button size="normal">普通按钮</h-button>
    <h-button size="small">小型按钮</h-button>
    <h-button size="mini">迷你按钮</h-button>
  </demo-block>
  <demo-block title="禁用状态">
    <h-button type="primary" disabled="disabled">禁用状态</h-button>
    <h-button type="danger" disabled="disabled">禁用状态</h-button>
  </demo-block>
  <demo-block title="加载状态">
    <h-button type="primary" loading="">loading</h-button>
    <h-button type="danger" loading="">loading</h-button>
  </demo-block>
</demo-section>
`;
